File: tests/fixtures/openai-cua/openai-cua-check-response-status-incomplete-fixed/checks_status_before_success.py
def run_step(client, input_items):
    response = client.responses.create(model="computer-use-preview", input=input_items)
    computer_calls = [item for item in response.output if item.type == "computer_call"]
    if not computer_calls:
        if response.status == "incomplete":
            return {"success": False, "reason": "truncated"}
        return {"success": True, "output": response.output_text}
    return {"success": False}

File: tests/fixtures/openai/openai-check-response-status-incomplete-fixed/checks-status-before-success.ts
export async function runTurn(client: any, model: string, input: any[]) {
  const response = await client.responses.create({ model, input, tools: [{ type: 'computer' }] });

  if (response.status === 'incomplete') {
    return { success: false, completed: false, error: 'truncated by token budget' };
  }

  const computerCalls = (response.output ?? []).filter((item: any) => item.type === 'computer_call');
  if (computerCalls.length === 0) {
    return { success: true, completed: true, final_message: response.output_text ?? '' };
  }

  return { success: true, completed: false };
}
